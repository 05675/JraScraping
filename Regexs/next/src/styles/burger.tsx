import styled from 'styled-components';

interface StyledBurgerProps {
  open?: boolean;
}

export const StyledBurger = styled.button`
  position: absolute;
  top: 50%;
  margin-top: -9px;
  right: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 18px;
  height: 18px;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;

  &:focus {
    outline: none;
  }

  div {
    width: 1.2rem;
    height: 0.13rem;
    background: #ffffff;
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    :first-child {
      transform: ${(props: StyledBurgerProps) => (props.open ? 'rotate(45deg)' : 'rotate(0)')};
    }

    :nth-child(2) {
      opacity: ${(props: StyledBurgerProps) => (props.open ? '0' : '1')};
      transform: ${(props: StyledBurgerProps) =>
        props.open ? 'translateX(15px)' : 'translateX(0)'};
    }

    :nth-child(3) {
      transform: ${(props: StyledBurgerProps) => (props.open ? 'rotate(-45deg)' : 'rotate(0)')};
    }
  }
`;
