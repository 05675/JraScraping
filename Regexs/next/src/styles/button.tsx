import styled, { css } from 'styled-components';
import { NenchoInsuranceStatus } from '@src/model/entity/nencho/nencho';

interface StyledButtonProps {
  important?: boolean;
  primary?: boolean;
  warning?: boolean;
  smallSize?: boolean;
  fontSize?: string;
  width?: string;
  height?: string;
  labelOffsetLeft?: string;
  noShadow?: boolean;
}

export const StyledButton = styled.button`
  display: inline-block;
  box-sizing:border-box;
  font-weight: bold;
  font-style: normal;
  line-height: 100%;
  text-align: center;
  background: #dedede;
  color: #b3b3b3;
  outline: none;
  border: none;
  border-radius: 24px;
  letter-spacing: 0.06em;
  padding-right:24px;
  box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
  cursor: pointer;
  cursor: hand;

  ${(props: StyledButtonProps) =>
    css`
      padding-left: ${props.labelOffsetLeft ?? '24px'};
      font-size: ${props.fontSize ?? '16px'};
      width: ${props.width ?? '100%'};
      height: ${props.height ?? '48px'};
    `}
  ${(props: StyledButtonProps) =>
    props.noShadow &&
    css`
      box-shadow: none;
    `}
  ${(props: StyledButtonProps) =>
    props.smallSize &&
    css`
      padding-left: 12px;
      padding-right: 12px;
      font-size: ${props.fontSize ?? '12px'};
      width: ${props.width ?? '100%'};
      height: ${props.height ?? '24px'};
      box-shadow: none;
    `}
  ${(props: StyledButtonProps) =>
    props.important &&
    css`
      background: #3aa0fa;
      color: #ffffff;
      :focus {
        background: #5cb1fc;
        color: #ffffff;
      }
    `}
  ${(props: StyledButtonProps) =>
    props.primary &&
    css`
      border: 2px solid #525252;
      box-sizing: border-box;
      background: #ffffff;
      color: #525252;
      :focus {
        background: #525252;
        color: #ffffff;
      }
    `}
  ${(props: StyledButtonProps) =>
    props.warning &&
    css`
      border: 2px solid #fa3939;
      box-sizing: border-box;
      background: none;
      color: #fa3939;
      :focus {
        background: #fa3939;
        color: #ffffff;
      }
    `}
  :disabled {
    outline: none;
    border: none;
    background: #dedede;
    color: #b3b3b3;
  }
`;

interface NenchoButtonProps extends StyledButtonProps {
  nenchoStatus?: number;
}

export const NenchoButton = styled(StyledButton)<NenchoButtonProps>`
  ${(props: NenchoButtonProps) =>
    props.nenchoStatus === NenchoInsuranceStatus.NOT_SUBMITTED &&
    css`
      animation: changeColor 1.47s 0.46s;
      @keyframes changeColor {
        0% {
          background-color: #3aa0fa;
          box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
          animation-timing-function: ease-out;
        }
        20% {
          background-color: #5cb1fc;
          box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);
        }
        88% {
          background-color: #5cb1fc;
          box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);
          animation-timing-function: ease-out;
        }
        100% {
          background-color: #3aa0fa;
          box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
        }
      }
    `}
`;
