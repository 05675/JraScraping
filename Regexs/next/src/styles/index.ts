export { StyledButton } from './button';
export { StyledInput } from './input';
export { StyledLabel } from './label';
export * from './listCommon';
