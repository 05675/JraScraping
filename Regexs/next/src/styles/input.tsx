import styled, { css } from 'styled-components';

export interface InputProps {
  type?: string;
  maxLength?: number;
  isError?: boolean;
  isPaddingRight?: boolean;
}

export const StyledInput = styled.input`
  -webkit-appearance: none;
  width: 100%;
  height: 48px;
  padding: 0;
  color: #525252;
  box-sizing: border-box;
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  line-height: 180%;
  background: rgba(0, 0, 0, 0);
  letter-spacing: 0.06em;
  border: none;
  border-radius: 0px;
  border-bottom: 1px solid #ebebeb;

  ${(props: InputProps) =>
    !props.isError &&
    css`
      :focus {
        outline: none;
        border-bottom: 1px solid #3aa0fa;
        box-shadow: 0px 1px 0px #3aa0fa;
      }
    `}

  ${(props: InputProps) =>
    props.isError &&
    css`
      border-bottom: 1px solid #fa3939;
      padding-right: 24px;
      :focus {
        outline: none;
        box-shadow: 0px 1px 0px #fa3939;
      }
    `}
  
  ${(props: InputProps) =>
    props.isPaddingRight &&
    css`
      padding-right: ${props.isError ? '54px' : '14px'};
    `} :disabled {
    color: #b3b3b3;
  }
`;
