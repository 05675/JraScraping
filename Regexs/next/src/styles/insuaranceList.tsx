import styled, { StyledComponent } from 'styled-components';
import React, { ReactElement } from 'react';

/**
 * リストのアイテムに格納するプロパティのインターフェース
 *
 * @interface InsuaranceItemProps
 */
interface InsuaranceItemProps {
  kind: string;
  firm: string;
  key: number;
}

/**
 * リストのプロパティのインターフェース
 *
 * @interface InsuarancesListProps
 */
export interface InsuarancesListProps {
  listItems: InsuaranceItemProps[];
}

/**
 * リスト全体のコントロール
 *
 * @param {InsuarancesListProps} props リストアイテムの配列
 * @returns {ReactElement} 表示する保険のリスト
 */
export const InsuaranceList: React.FC<InsuarancesListProps> = props => {
  return (
    <>
      {props.listItems.map(insuarance => {
        return (
          <InsuaranceListItem kind={insuarance.kind} firm={insuarance.firm} key={insuarance.key} />
        );
      })}
    </>
  );
};

/**
 * リストの1つ1つのアイテム
 *
 * @param {InsuaranceItemProps} props 保険会社名と保険種別
 * @returns {ReactElement} 表示するリストのアイテム
 */
const InsuaranceListItem: React.FC<InsuaranceItemProps> = props => {
  return (
    <StyledInsuaranceListItem>
      <StyledNameFirmWrapper theme={{ size: 11 }}>
        <div>{props.kind}</div>
        <StyledFirmName>{props.firm}</StyledFirmName>
      </StyledNameFirmWrapper>
      <StyledArrowWrapper theme={{ size: 1 }}>→</StyledArrowWrapper>
    </StyledInsuaranceListItem>
  );
};

/**
 *リストの中のアイテム全体のデザイン
 *
 * @returns {StyledComponent} コントロール
 */
const StyledInsuaranceListItem = styled.div`
  display: flex;
  border: solid #ebebeb;
  border-width: 1px 0px 1px 0px;
  padding: 10px 16px 9px 16px;
`;

/**
 * 保険会社名を包含するコントロールのデザイン
 *
 * @returns {StyledComponent} コントロール
 */
const StyledNameFirmWrapper = styled.div`
  flex: ${({ theme }) => theme.size};
`;

/**
 * 保険種別を包含するコントロールのデザイン
 *
 * @returns {StyledComponent} コントロール
 */
const StyledFirmName = styled.div`
  font-size: 12px;
  color: #797979;
  margin-top: 18px;
`;

/**
 * 矢印を包含するコントロールのデザイン
 *
 * @returns {StyledComponent} コントロール
 */
const StyledArrowWrapper = styled(StyledNameFirmWrapper)`
  flex: ${({ theme }) => theme.size};
  display: flex;
  justify-content: center;
  align-items: center;
`;
