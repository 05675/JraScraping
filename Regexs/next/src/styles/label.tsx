import styled from 'styled-components';

export const StyledLabel = styled.label`
  font-size: 12px;
  font-style: normal;
  font-weight: normal;
  color: #797979;
  line-height: 150%;
  letter-spacing: 0.06em;
`;

export const StyledTitle = styled.title`
  display: block;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 100%;
  text-align: center;
  color: #525252;
  padding-top: 20px;
  padding-bottom: 28px;
  width: 100%;
`;

export const StyledText = styled.div`
  font-size: 12px;
  font-style: normal;
  font-weight: normal;
  color: #525252;
  line-height: 150%;
  letter-spacing: 0.06em;
`;

export const StyledErrorText = styled.div`
  font-size: 12px;
  font-style: normal;
  font-weight: normal;
  color: #fa3939;
  line-height: 150%;
  letter-spacing: 0.06em;
`;
