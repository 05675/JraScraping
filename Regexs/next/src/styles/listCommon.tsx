import styled from 'styled-components';

export const StyledUl = styled.ul`
  padding: 16px;
  margin: 0;
`;
export const StyledList = styled.li`
  display: inline-block;
  position: relative;
  width: 100%;
  height: 88px;
  margin-bottom: 9px;
  padding: 0;
  background: #ffffff;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.2);
  border-radius: 8px;
  cursor: pointer;
  cursor: hand;
`;
export const StyledListItemTitle = styled.span`
  font-style: normal;
  font-size: 16px;
  line-height: 100%;
  letter-spacing: 0.06em;
  font-weight: normal;
  color: #525252;
  vertical-align: top;
`;
export const StyledListItemStatus = styled.h5`
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 100%;
  letter-spacing: 0.06em;
  margin: 0;
`;
export const StyledListItemDueDate = styled.h5`
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 100%;
  letter-spacing: 0.06em;
  margin: 0;
`;
export const StyledListItemArrow = styled.svg`
  width: 13.41px;
  height: 13.41px;
  margin: 0;
  padding: 0;
`;

export const StyledListNencho = styled.ul`
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0 16px;
  font-size: 0;
  list-style: none;
  box-sizing: border-box;
  min-height: 100%;
`;

export const StyledListItemTitleNencho = styled.div`
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 100%;
  letter-spacing: 0.06em;
  color: #525252;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const StyledListItemLabelNencho = styled.h5`
  width: 100%;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 100%;
  letter-spacing: 0.06em;
  color: #949494;
  margin: 0;
  padding: 0;
`;
