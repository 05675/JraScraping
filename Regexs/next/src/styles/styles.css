body {
  margin: 0;
  font-family: Noto Sans JP, Hiragino Sans, Meiryo, sans-serif;
  background-color: #edf4fa;
}

button,
input,
select,
textarea {
  font-family: inherit;
}

.common-header {
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
  height: 48px;
  position: fixed;
}

.page-header-task {
  top: 48px;
  left: 0;
  right: 0;
  z-index: 1;
  background: #edf4fa;
  height: 72px;
  position: fixed;
}

.page-header-life-insurance {
  top: 48px;
  left: 0;
  right: 0;
  z-index: 1;
  position: fixed;
}
