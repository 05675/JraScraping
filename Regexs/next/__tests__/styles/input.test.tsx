import { StyledInput } from '@src/styles/input';
import React from 'react';
import renderer from 'react-test-renderer';

describe('StyledInput', () => {
  test('snapshot isError isPaddingRight', () => {
    const content = renderer
      .create(<StyledInput isError={false} isPaddingRight={false} />)
      .toJSON();
    expect(content).toMatchSnapshot();
  });
  test('snapshot isError isNotPaddingRight', () => {
    const content = renderer.create(<StyledInput isError={false} isPaddingRight />).toJSON();
    expect(content).toMatchSnapshot();
  });
  test('snapshot isNotError isPaddingRight', () => {
    const content = renderer.create(<StyledInput isError isPaddingRight={false} />).toJSON();
    expect(content).toMatchSnapshot();
  });
  test('snapshot isNotError isNotPaddingRight', () => {
    const content = renderer.create(<StyledInput isError isPaddingRight />).toJSON();
    expect(content).toMatchSnapshot();
  });
});
