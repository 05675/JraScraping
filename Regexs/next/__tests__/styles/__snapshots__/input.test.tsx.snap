// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput snapshot isError isNotPaddingRight 1`] = `
<input
  className="sc-AxjAm eGfAp"
/>
`;

exports[`StyledInput snapshot isError isPaddingRight 1`] = `
<input
  className="sc-AxjAm dmWriN"
/>
`;

exports[`StyledInput snapshot isNotError isNotPaddingRight 1`] = `
<input
  className="sc-AxjAm ljofMs"
/>
`;

exports[`StyledInput snapshot isNotError isPaddingRight 1`] = `
<input
  className="sc-AxjAm ktMOEw"
/>
`;
